README.md

```diff
@@ -22,21 +22,10 @@ LIBS := $(UK_LIBS)/lib-newlib:$(UK_LIBS)/lib-libsodium
 
 ## Testing:
 
-To run the tests, select the `Enable libsodium tests` option in menuconfig.
-This will allow you to include `sodium/uk_sodium_test.h` and call
-`uk_sodium_run_tests()`. A testing application can be as simple as:
-```
-#include <stdio.h>
-#include <sodium/uk_sodium_test.h>
-
-int main(int argc, char *argv[])
-{
-	(void)argc;
-	(void)argv;
+To run the tests, select the `Enable libsodium tests` option in menuconfig. Alternatively you can
+select the `Enable all tests across all libraries` option under the configuration of the `uktest`
```

Comment:
This didn't work for me. If I use the `Enable all tests across all libraries` option, tests are indeed built (I can see the symbols with `nm build/libsodium.o | grep test`) but they are not run.

Reply:
I can reproduce but not sure why this is happening. I'll put higher priority on the SMP PR though, and if time permits I'll have a closer look.

Reply:
Let's leave it like this at this point. We can add an issue and figure that out later. I'll mark it as approved for this release.

Reply:
I think the issue comes from the `ifdef` guard at [line 804](https://github.com/unikraft/lib-libsodium/blob/ac9e7db5117e50667fe43e75bf67f9dd4212b7c3/tests/test_minimal.c#L804).

When the `Enable all tests across all libraries` option is selected, the `CONFIG_LIBSODIUM_TEST_MINIMAL` config option is not set, since it will create a circular dependency, that's why the `ifneq` workaround is required in the Makefile. See the discussion [here](https://github.com/unikraft/unikraft/pull/431#discussion_r835813739).

The test running never happens because the testsuite registration inside the guard mentioned above is not executed. I think the guard should be eliminated since the tests are never built if their execution is not intended.

Reply:
Thanks, @StefanJum . @michpappas , see if it makes sense to remove the guard. I will approve the PR in any situation. If there isn't a clean solution, we'll merge it as it is and leave it for later to find a proper solution.